File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = [1] * len(nums)
        suffix = [1] * len(nums)

        product = 1

        for i in range(len(nums)):
            prefix[i] = product
            product = product * nums[i]
        
        product = 1

        for i in range(len(nums) - 1, -1, -1):
            suffix[i] = product
            product = product * nums[i]

        result = []

        for i in range(len(nums)):
            result.append(prefix[i] * suffix[i])
        
        return result
            
        

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cs
public class Solution {
    public int FindMin(int[] nums) {
        
        int l = 0;
        int r = nums.Length - 1;
        int result = nums[0];

        while (l <= r)
        {
            if (nums[l] <= nums[r])
            {
                result = Math.Min(result, nums[l]);
                break;
            }
            else
            {
                int mid = (l + r) / 2;

                if (nums[mid] >= nums[l])
                {
                    result = Math.Min(nums[l], result);
                    l = mid + 1;
                }
                else
                {
                    result = Math.Min(nums[mid], result);
                    r = mid - 1;
                }
            }
        }

        return result;
    }
}
